pkg/audit/verifier.go

```diff
@@ -489,24 +517,53 @@ func (verifier *Verifier) GetShare(ctx context.Context, limit *pb.AddressedOrder
 	}, nil
 }
 
+// RemoveFailedPieces removes lost pieces from a pointer
+func (verifier *Verifier) RemoveFailedPieces(ctx context.Context, path string, pointer *pb.Pointer, failedNodes storj.NodeIDList) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	if len(path) == 0 {
+		return nil
+	}
+
+	remoteSegment := pointer.GetRemote()
+	newRemotePieces := remoteSegment.RemotePieces[:0]
+	for _, piece := range remoteSegment.RemotePieces {
+		isFailed := false
+		for _, failedNode := range failedNodes {
+			if piece.NodeId == failedNode {
+				isFailed = true
+				break
+			}
+		}
+		if !isFailed {
+			newRemotePieces = append(newRemotePieces, piece)
```

Comment:
Can you replace line 534 (`isFailed = true`) with this line? Then you should be able to get rid of `isFailed` completely.

Reply:
fixing